src/libs/actions/User.ts

```diff
@@ -469,6 +471,101 @@ function triggerNotifications(onyxUpdates: OnyxServerUpdate[]) {
     });
 }
 
+const isChannelMuted = (reportId: string) =>
+    new Promise((resolve) => {
+        const connectionId = Onyx.connect({
+            key: `${ONYXKEYS.COLLECTION.REPORT}${reportId}`,
+            callback: (report) => {
+                Onyx.disconnect(connectionId);
+
+                resolve(
+                    !report?.notificationPreference ||
+                        report?.notificationPreference === CONST.REPORT.NOTIFICATION_PREFERENCE.MUTE ||
+                        report?.notificationPreference === CONST.REPORT.NOTIFICATION_PREFERENCE.HIDDEN,
+                );
+            },
+        });
+    });
+
+function playSoundForMessageType(pushJSON: OnyxServerUpdate[]) {
+    const reportActionsOnly = pushJSON.filter((update) => update.key.includes('reportActions_'));
```

Comment:
There's case of `key` not existing in `update`, which lead to https://github.com/Expensify/App/issues/36632.
i.e. CloseAccount pusher event